$OpenBSD: patch-ev_c,v 1.2 2012/08/14 07:11:09 jasper Exp $

Add support for OpenBSD/alpha and OpenBSD/mips64 (latter from ev.c -r1.412).

--- ev.c.orig	Sat Feb  4 20:09:52 2012
+++ ev.c	Fri Jul 27 14:30:52 2012
@@ -579,6 +579,10 @@ struct signalfd_siginfo
       #define ECB_MEMORY_FENCE_RELEASE __asm__ __volatile__ ("membar #LoadStore |             #StoreStore")
     #elif defined(__s390__) || defined(__s390x__)
       #define ECB_MEMORY_FENCE         __asm__ __volatile__ ("bcr 15,0" : : : "memory")
+    #elif defined(__mips__)
+      #define ECB_MEMORY_FENCE         __asm__ __volatile__ ("sync" : : : "memory")
+    #elif defined(__alpha__)
+      #define ECB_MEMORY_FENCE         __asm__ __volatile__ ("mb" : : : "memory")
     #endif
   #endif
 #endif
